frontend/public/components/secrets/create-secret.tsx

```diff
@@ -59,116 +84,258 @@ class BaseEditSecret_ extends SafetyFirst<BaseEditSecretProps_, BaseEditSecretSt
+        ? <BasicAuthSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData}/>
+        : <SSHAuthSubform onChange={this.onDataChanged.bind(this)} stringData={this.state.stringData}/>
+      }
+    </React.Fragment>;
+  }
+}
+
+const secretFormFactory = secretType => {
+  return secretType === 'webhook' ? withSecretForm(WebHookSecretForm) : withSecretForm(SourceSecretForm);
+};
+
+class BasicAuthSubform extends React.Component<BasicAuthSubformProps, BasicAuthSubformState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      username: this.props.stringData.username || '',
+      password: this.props.stringData.password || '',
+    };
+    this.changeData = this.changeData.bind(this);
+  }
+  changeData(event) {
+    this.setState({
+      [event.target.name]: event.target.value
+    }, () => this.props.onChange(this.state));
+  }
+  render() {
+    return <React.Fragment>
+      <div className="form-group">
+        <label className="control-label" htmlFor="username">Username</label>
+        <div className="modal-body__field">
+          <input className="form-control"
+            id="username"
+            aria-describedby="username-help"
+            type="text"
+            name="username"
+            onChange={this.changeData}
+            value={this.state.username}
+            required />
+          <p className="help-block" id="username-help">Optional username for Git authentication.</p>
+        </div>
+      </div>
+      <div className="form-group">
+        <label className="control-label" htmlFor="password">Password or Token</label>
+        <div className="modal-body__field">
+          <input className="form-control"
+            id="password"
+            aria-describedby="password-help"
+            type="password"
+            name="password"
+            onChange={this.changeData}
+            value={this.state.password}
+            required />
+          <p className="help-block" id="password-help">Password or token for Git authentication. Required if a ca.crt or .gitconfig file is not specified.</p>
+        </div>
+      </div>
+    </React.Fragment>;
+  }
+}
+
+class SSHAuthSubform extends React.Component<SSHAuthSubformProps, SSHAuthSubformState> {
+  constructor (props) {
+    super(props);
+    this.state = {
+      'ssh-privatekey': this.props.stringData['ssh-privatekey'] || '',
+    };
+    this.changeData = this.changeData.bind(this);
+  }
+  changeData(event) {
+    this.setState({
+      'ssh-privatekey': event.target.value
+    }, () => this.props.onChange(this.state));
+  }
+  onFileChange(fileData) {
+    this.setState({
+      'ssh-privatekey': fileData
+    }, () => this.props.onChange(this.state));
+  }
+  render() {
+    return <div className="form-group">
+      <label className="control-label" htmlFor="ssh-privatekey">SSH Private Key</label>
+      <div className="modal-body__field">
+        <FileInput onChange={this.onFileChange.bind(this)}/>
```

Comment:
Prefer to bind in constructor.

nit: space before `/` in self-closing tags since that's the style used throughout the existing code